drizzlepac/outputimage.py

```diff
@@ -446,31 +446,38 @@ def writeFITS(self, template, sciarr, whtarr, ctxarr=None,
 
             # explicitly set EXTEND to FALSE for simple FITS files.
             dim = len(sciarr.shape)
-            hdu.header.set('extend', value=False, after='NAXIS%s'%dim)
+            hdu.header.set('extend', value=False, after='NAXIS%s' % dim)
+
+            # explicitly remove EXTNAME, EXTVER from header - they are not allowed
+            for kw in RESERVED_KEYS[-2:]:
+                if kw in hdu.header:
+                    del hdu.header[kw]
```

Comment:
I do not think this is necessary at all because `'EXTNAME'` and `'EXTVER'` should not be present in `hdu` for the following reasons:
1. On line `#423` `hdu_header` is defined as a copy of a primary HDU header (`hdu_header = prihdu.header.copy()`) so it cannot (or should not) have `'EXTNAME'` and `'EXTVER'` in it.
2. On lines `#441-444`, `hdu` is created as, i.e., `hdu = fits.ImageHDU(data=sciarr, header=hdu_header)` without `name` or `ver` specified and hence, even though it inherits from `hdu_header`, it also will not contain `'EXTNAME'` and `'EXTVER'`.

So, this code seems unnecessary. But if you do want to keep it, I would suggest the following modification:

```suggestion
            for kw in ['EXTNAME', 'EXTVER']:
                hdu.header.pop(kw, None)
```
Explicitly specifying keywords to be deleted is safer if anyone later would like to update `RESERVED_KEYS` constant and add other words after the last one in the list.

Reply:
Moreover, line `#444` (`hdu = fits.ImageHDU(data=sciarr, header=hdu_header)`) could be replaced directly with 

```python
hdu = fits.PrimaryHDU(data=sciarr, header=hdu_header)
```
which I highly recommend.

Reply:
All these concerns were addressed in 4b565a7, including using PrimaryHDU instead.  Unfortunately, this code needs to remain in place (as noted in the new comments) since the HDU that gets created MAY be a CompImageHDU instead of a PrimaryHDU depending on whether 'driz_sep_compress' is turned on or not.